buildscripts/CommonAssemblyInfo.cs

```diff
@@ -12,12 +12,41 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using System;
+using System.Runtime.InteropServices;
 using System.Reflection;
+using System.Security;
+
+[assembly: AssemblyProduct(
+	#if DOTNET45 // order is important
+	"Castle Core for .NET Framework 4.5"
+	#elif DOTNET40
+	"Castle Core for .NET Framework 4.0"
+	#elif DOTNET35
+	"Castle Core for .NET Framework 3.5"
+	#else
+	"Castle Core for .NET Core 1.0"
+	#endif
+)]
 
 [assembly: AssemblyCompany("Castle Project")]
-[assembly: AssemblyCopyright("Copyright (c) 2004-2014 Castle Project - http://www.castleproject.org")]
-#if ASSEMBLY_CLSCOMPLIANT_FALSE
-[assembly: System.CLSCompliant(false)]
-#else
-[assembly: System.CLSCompliant(true)]
+[assembly: AssemblyCopyright("Copyright (c) 2004-2016 Castle Project - http://www.castleproject.org")]
+
+[assembly: AssemblyVersion("3.3.0")]
+[assembly: AssemblyFileVersion("3.3.4")]

```

Comment:
Looks good. For ease of future maintenance, I'd consider

1. pulling version numbers out of Settings.proj and having the Get-NuGetVariables read from this file, and
2. maybe making the major/minor/update components into consts so they can be defined once

These things are definitely "bonus".

One thing we do lose, though, is the BuildName in the `AssemblyInformationalVersion`. Comes from TeamCity's Build_Number, I think. This was one of the things that I wasn't sure whether it was important to you.